core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1175,7 +1186,10 @@ private Optional<RetentionSizeData> buildRetentionSizeData(long retentionSize,
                         }
                     }
                 }
-                brokerTopicStats.topicStats(topicIdPartition.topic()).recordRemoteLogSizeComputationTime(topicIdPartition.partition(), time.milliseconds() - startTimeMs);
+
+                BrokerTopicMetrics brokerTopicMetrics = brokerTopicStats.topicStats(topicIdPartition.topic());
+                brokerTopicMetrics.recordRemoteLogSizeComputationTime(topicIdPartition.partition(), time.milliseconds() - startTimeMs);
+                brokerTopicStats.allTopicsStats().recordRemoteLogSizeComputationTime(topicIdPartition.topic(), brokerTopicMetrics.remoteLogSizeComputationTime());
```

Comment:
could go in a single function RequestHandler with name updateRemoteLogSizeComputationTime(), similar to existing updateBytesOut(). This will ensure that in future folks don't forget to update allTopics if they update the broekrTopic metric.